packages/firestore/src/platform/browser/webchannel_connection.ts

```diff
@@ -241,11 +241,15 @@ export class WebChannelConnection extends RestConnection {
     // on a closed stream
     let closed = false;
 
+    let onOpenTimestamp: Date | null = null;
+    onOpenTimestamp = new Date();
+
     const streamBridge = new StreamBridge<Req, Resp>({
       sendFn: (msg: Req) => {
         if (!closed) {
           if (!opened) {
             logDebug(LOG_TAG, 'Opening WebChannel transport.');
+            // onOpenTimestamp = new Date();  // This isn't working anymore. Hmm...
```

Comment:
I think the difference in timing here is negligible. Line 252 would fire very quickly after line 245. We always send a message one we open the stream. The only exception is possibly a user adding and removing a snapshot listener in quick succession, which might cause us to open a stream and then deciding that we don't need it. I would have to convince myself that this is an actual problem and am mostly indifferent on whether we set the timer here or earlier.